src/rpc/client.cpp

```diff
@@ -214,15 +214,25 @@ CRPCConvertTable::CRPCConvertTable()
 
 static CRPCConvertTable rpcCvtTable;
 
+static bool IsHexDigit(const char c) {
+    return ('0' <= c && c <= '9') || ('a' <= c && c <= 'f') || ('A' <= c && c <= 'F');
+}
+
+static bool IsHashVal(const std::string& str) {
+    return (str.length() == 64) && std::all_of(str.begin(), str.end(), IsHexDigit);
+}
+
 /** Non-RFC4627 JSON parser, accepts internal values (such as numbers, true, false, null)
```

Comment:
Maybe add that this method also accepts tx and blockhashes?
```suggestion
/** Non-RFC4627 JSON parser, accepts internal values (such as numbers, true, false, null or tx and blockhashes)
```

Reply:
makes sense, I amended the comment.

Reply:
Oh, I forgot: You probably want to change this in `src/rpc/client.h` as well for consistency :)

Reply:
Done